components/lifesensor_macro/include/macro_task.h

```diff
@@ -0,0 +1,72 @@
+#ifndef __MACRO_TASK__
+#define __MACRO_TASK__
+
+#include <freertos/task.h>
+
+#define MACRO_DEFAULT_STACK_SIZE 0x1000
+/**
+ * struct lifesensor_task_t - generic task struct
+ * @tcb:    Holds the FreeRTOS tcb.
+ * @handle: Provides a reference to the task.
+ * @stack:  Stores the address of the tasks' stack.
+ *
+ * This struct serves as an abstraction for operating with FreeRTOS tasks such
+ * that it includes necessary fields for managing a tasks data. It is intended
+ * to be used together with the LIFESENSOR_TASK() macro.
+ */
+typedef struct {
+	StaticTask_t tcb;
+	TaskHandle_t handle;
+	StackType_t stack[];
+} lifesensor_task_t;
+
+
+/**
+ * LIFESENSOR_TASK() - Generic task struct declaration macro
+ * @stacksize:  Determines the stack depth used.
+ *
+ * The macro is used for easier task struct allocation. The resulting
+ * task struct should be initialized through the corresponding 
+ * `lifesensor_task_init()` function macro.
+ *
+ * NOTE:	`task` houses the actual task struct of type `lifesensor_task_t`.
+ * 			`lifesensor_task_t`s stack attribute of undefined length will
+ * 			point to the stack declared here with size `stacksize`.
+ *
+ */
+#define LIFESENSOR_TASK(stacksize)                                          \
+	struct {                                                                \
+		lifesensor_task_t task;                                             \
+		StackType_t stack[stacksize];                                       \
+	}
+
+
+/**
+ * lifesensor_task_init() - Initialize a task generated with LIFESENSOR_TASK
+ * @anon_task:	Pass a pointer to instance created with LIFESENSOR_TASK.
+ * @name:		Name the task to recognize it later.
+ * @func:		Provide a function pointer to execute in task.
```

Comment:
```suggestion
 * @func:		Points to function pointer used as entry point of the task.
```
there can only be one